api/catalog_test.go

```diff
@@ -132,25 +129,23 @@ func TestCatalog_Services(t *testing.T) {
 	defer s.Stop()
 
 	catalog := c.Catalog()
-
-	if err := testutil.WaitForResult(func() (bool, error) {
+	for r := retry.OneSec(); r.NextOr(t.FailNow); {
 		services, meta, err := catalog.Services(nil)
 		if err != nil {
-			return false, err
+			t.Log(err)
+			continue
 		}
-
 		if meta.LastIndex == 0 {
-			return false, fmt.Errorf("Bad: %v", meta)
+			t.Logf("Bad: %v", meta)
```

Comment:
We also have some "Bad" and "bad" variations - might be good to make these all one way or the other (I'd personally do "bad".